wrapped-bridged-usdc/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+#![no_std]
+
+elrond_wasm::imports!();
+
+#[elrond_wasm::contract]
+pub trait WrappedBridgedUsdc {
+    #[init]
+    fn init(&self, universal_bridged_usdc_token_id: TokenIdentifier) {
+        self.universal_bridged_usdc_token_id()
+            .set(universal_bridged_usdc_token_id);
+    }
+
+    #[only_owner]
+    #[endpoint(whitelistUsdc)]
+    fn whitelist_usdc(&self, chain_specific_usdc_token_id: TokenIdentifier) {
+        self.chain_specific_usdc_token_ids()
+            .insert(chain_specific_usdc_token_id);
+    }
+
+    #[only_owner]
+    #[endpoint(blacklistUsdc)]
+    fn blacklist_usdc(&self, chain_specific_usdc_token_id: TokenIdentifier) {
+        self.chain_specific_usdc_token_ids()
+            .swap_remove(&chain_specific_usdc_token_id);
+    }
+
+    // endpoints
+
+    #[payable("*")]
+    #[endpoint(wrapUsdc)]
+    fn wrap_usdc(
+        &self,
+        #[payment_token] payment_token: TokenIdentifier,
+        #[payment_amount] payment_amount: BigUint,
+    ) {
+        let chain_specific_usdc_token_id = &payment_token;
+        require!(
+            self.chain_specific_usdc_token_ids()
+                .contains(chain_specific_usdc_token_id),
+            "Wrong esdt token"
+        );
+
+        let universal_bridged_usdc_token_id = self.universal_bridged_usdc_token_id().get();
+        require!(payment_amount > 0u32, "Payment must be more than 0");
+
+        self.send()
+            .esdt_local_mint(&universal_bridged_usdc_token_id, 0, &payment_amount);
+
+        let caller = self.blockchain().get_caller();
```

Comment:
I think it would be better to keep in storage the amount of chain_specific_usdc_token_id deposited through this function. As there can be some cases when the universal_bridged_usdc_token_id is sent back to the contract and the contract does not know about that. (esdtTransfer fails cross shard and gets returned to the contract). This will lead to some confusion on metrics and explanations forever and ever.